src/pj_internal.c

```diff
@@ -134,36 +136,193 @@ size_t pj_strlcpy(char *dst, const char *src, size_t siz) {
  * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
  * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
  * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
 
-  Source: http://www.i-pi.com/Training/EthicalHacking/Solutions/strlcpy.c
+  Source: http://cvsweb.openbsd.org/cgi-bin/cvsweb/src/lib/libc/string/strlcpy.c
```

Comment:
As long as we stick with C89, we can't assume having `strncpy`. It's only available [since C99](http://en.cppreference.com/w/c/string/byte/strncpy) and the compiler will not make it available when we pass `-std=c89`, regardless of whether it is supported.

Reply:
@aaronpuchert, I think your statement about strncpy if wrong: if I read the page at the end of the link you provide correctly, the prototype simply changed to include the 'restrict' qualifier with the publication of the C99 standard.

Reply:
strncpy is in C89/90.

https://stackoverflow.com/questions/17014835/where-can-i-find-the-c89-c90-standards-in-pdf-format
http://port70.net/~nsz/c/c89/c89-draft.txt is a 

    4.11.2.4 The strncpy function

    Synopsis

         #include <string.h>
         char *strncpy(char *s1, const char *s2, size_t n);

http://pubs.opengroup.org/onlinepubs/9699919799/functions/strncpy.html

    CHANGE HISTORY
        First released in Issue 1. Derived from Issue 1 of the SVID.

    Issue 6
        The strncpy() prototype is updated for alignment with the ISO/IEC 9899:1999 standard.

    Issue 7
        The stpncpy() function is added from The Open Group Technical Standard, 2006, Extended API Set Part 1.

Reply:
@schwehr thanks for a link to an excellent resource - diving into it, it took med 3 hours to get back to the surface :-)

I will drop my planned effort of removing all calls to `strncpy`.

Reply:
You are right, I was a bit quick to judge.